vnext/Microsoft.ReactNative/Views/TouchEventHandler.cpp

```diff
@@ -432,27 +458,37 @@ void TouchEventHandler::DispatchTouchEvent(TouchEventType eventType, size_t poin
   } else
 #endif // USE_FABRIC
   {
-    auto touches = folly::dynamic::array();
+    winrt::Microsoft::ReactNative::JSValueArray touches;
     for (const auto &pointer : m_pointers) {
-      folly::dynamic touch = GetPointerJson(pointer, pointer.target);
-      touches.push_back(touch);
+      touches.push_back(GetPointerJson(pointer, pointer.target));
     }
 
     // Package up parameters and invoke the JS event emitter
-    const char *eventName = GetTouchEventTypeName(eventType);
+    const wchar_t *eventName = GetTouchEventTypeName(eventType);
     if (eventName == nullptr)
       return;
-    folly::dynamic params = folly::dynamic::array(eventName, std::move(touches), std::move(changedIndices));
 
-    m_context->CallJSFunction("RCTEventEmitter", "receiveTouches", std::move(params));
+    const auto paramsWriter = MakeJSValueArgWriter(eventName, std::move(touches), std::move(changedIndices));
+    if (eventType == TouchEventType::Move || eventType == TouchEventType::PointerMove) {
+      BatchingEmitter().EmitCoalescingJSEvent(
+          L"RCTEventEmitter",
+          L"receiveTouches",
+          std::move(eventName),
+          m_pointers[pointerIndex].pointerId,
+          paramsWriter);
+    } else {
+      BatchingEmitter().EmitJSEvent(L"RCTEventEmitter", L"receiveTouches", paramsWriter);
+    }
   }
 }
 
 bool TouchEventHandler::DispatchBackEvent() {
   if (m_context->State() != Mso::React::ReactInstanceState::Loaded)
     return false;
 
-  m_context->CallJSFunction("RCTDeviceEventEmitter", "emit", folly::dynamic::array("hardwareBackPress"));
+  BatchingEmitter().EmitJSEvent(
+      L"RCTDeviceEventEmitter", L"emit", winrt::Microsoft::ReactNative::MakeJSValueArgWriter(L"hardwardBackPress"));
```

Comment:
nit: the `winrt::Microsoft::ReactNative` can be replaced with `React`. It is not worth a new iteration. Just FYI. The `React` is an alias for `winrt::Microsoft::ReactNative`.

Reply:
I was also confused why I needed the namespace here, but not above... thoughts @vmoroz?

Reply:
@rozele I think it is due to the [ADL](https://en.cppreference.com/w/cpp/language/adl).